source/channel_bootstrap.c

```diff
@@ -1031,8 +1056,9 @@ static inline int s_setup_server_tls(struct server_connection_args *connection_a
         return AWS_OP_ERR;
     }
 
-    tls_handler =
-        aws_tls_server_handler_new(connection_args->bootstrap->allocator, &connection_args->tls_options, tls_slot);
+    struct aws_tls_connection_options tls_options = connection_args->tls_options;
```

Comment:
add comment: /* Shallow-copy tls_options so we can override the user_data, making it specific to this channel */